aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java

```diff
@@ -0,0 +1,178 @@
+import com.redhat.cloud.notifications.db.EmailAggregationResources;
+import com.redhat.cloud.notifications.db.ResourceHelpers;
+import com.redhat.cloud.notifications.models.AggregationCommand;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.smallrye.reactive.messaging.connectors.InMemoryConnector;
+import io.smallrye.reactive.messaging.connectors.InMemorySink;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junitpioneer.jupiter.SetSystemProperty;
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static com.redhat.cloud.notifications.models.EmailSubscriptionType.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@QuarkusTestResource(TestLifecycleManager.class)
+class DailyEmailAggregationJobTest {
+
+    @Inject
+    ResourceHelpers helpers;
+
+    @Inject
+    DailyEmailAggregationJob testee;
+
+    @Inject
+    @Any
+    InMemoryConnector connector;
+
+    @BeforeEach
+    void setUp() {
+        helpers.purgeAggregations();
+        helpers.purgeEmailSubscriptions();
+    }
+
+    @AfterEach
+    void tearDown() {
+        helpers.purgeAggregations();
+        helpers.purgeEmailSubscriptions();
+    }
+
+    @Test
+    @SetSystemProperty(key = "notifications.aggregator.email.subscription.periodic.cron.enabled", value = "true")
+    void shouldSentTwoAggregationsToKafkaTopic() {
+        helpers.addEmailAggregation("tenant", "rhel", "policies", "somePolicyId", "someHostId");
+        helpers.addEmailAggregation("tenant", "rhel", "unknown-application", "somePolicyId", "someHostId");
+
+        testee.processDailyEmail();
+
+        InMemorySink<String> results = connector.sink("aggregation");
+
+        assertEquals(2, results.received().size());
+
+        final String firstAggregation = results.received().get(0).getPayload();
+        assertTrue(firstAggregation.contains("accountId\":\"tenant"));
+        assertTrue(firstAggregation.contains("bundle\":\"rhel"));
+        assertTrue(firstAggregation.contains("application\":\"policies"));
+        assertTrue(firstAggregation.contains("subscriptionType\":\"DAILY"));
+
+        final String secondAggregation = results.received().get(1).getPayload();
+        assertTrue(secondAggregation.contains("accountId\":\"tenant"));
+        assertTrue(secondAggregation.contains("bundle\":\"rhel"));
+        assertTrue(secondAggregation.contains("application\":\"unknown-application"));
+        assertTrue(secondAggregation.contains("subscriptionType\":\"DAILY"));
+    }
+
+    @Test
+    void shouldNotChangeSomethingWhenCreatingSubscription() {
```

Comment:
Why does the `src/test` folder still contain code related to `EmailSubscription` while there isn't a trace of `EmailSubscription` or `endpoint_email_subscriptions` in `src/main`? It looks like all code related to `EmailSubscription` should be entirely removed from this module, but maybe I missed some place in `src/main` where it is actually needed.

Reply:
Right now it's needed for the tests only. For creating test data and so on.

Reply:
Yes I saw that, but why would we need that data if it's never used at run time?

Reply:
Actually, the entire `com.redhat.cloud.notifications.models` package (as well as all related code) from `src/test` seems removable.

Reply:
That's what I did with the latest 2-3 commits :)

Reply:
the test I copied from backend was a bit misleading as it mixed subscription and aggregation features withint a large test and I got confused by it.